bin/lib/rust_library_builder.py

```diff
@@ -0,0 +1,453 @@
+        rustbinpath = os.path.dirname(compilerexe)
+        rustpath = os.path.dirname(rustbinpath)
+        extraflags = ''
+        libcxx = ''
+
+        with open_script(Path(sourcefolder) / "build.sh") as f:
+            f.write('#!/bin/sh\n\n')
+
+            f.write(f'RUSTPATH={rustpath}\n')
+            f.write(f'CARGO={rustbinpath}/cargo\n')
+            f.write(f'PATH={rustbinpath}\n')
+
+            for line in self.buildconfig.prebuild_script:
+                f.write(f'{line}\n')
+
+            if self.buildconfig.build_type == "cargo":
+                cargoline = f'$CARGO build --target-dir {buildfolder} >buildlog.txt 2>&1\n'
+                f.write(cargoline)
+            else:
+                raise RuntimeError('Unknown build_type {self.buildconfig.build_type}')
+
+        self.setCurrentConanBuildParameters(buildos, buildtype, compilerType, compiler, libcxx, arch, stdver,
+                                            extraflags)
+
+    def setCurrentConanBuildParameters(self, buildos, buildtype, compilerTypeOrGcc, compiler, libcxx, arch, stdver,
+                                       extraflags):
+        self.current_buildparameters_obj['os'] = buildos
+        self.current_buildparameters_obj['buildtype'] = buildtype
+        self.current_buildparameters_obj['compiler'] = compilerTypeOrGcc
+        self.current_buildparameters_obj['compiler_version'] = compiler
+        self.current_buildparameters_obj['libcxx'] = libcxx
+        self.current_buildparameters_obj['arch'] = arch
+        self.current_buildparameters_obj['stdver'] = stdver
+        self.current_buildparameters_obj['flagcollection'] = extraflags
+        self.current_buildparameters_obj['library'] = self.libid
+        self.current_buildparameters_obj['library_version'] = self.target_name
+
+        self.current_buildparameters = ['-s', f'os={buildos}',
+                                        '-s', f'build_type={buildtype}',
+                                        '-s', f'compiler={compilerTypeOrGcc}',
+                                        '-s', f'compiler.version={compiler}',
+                                        '-s', f'compiler.libcxx={libcxx}',
+                                        '-s', f'arch={arch}',
+                                        '-s', f'stdver={stdver}',
+                                        '-s', f'flagcollection={extraflags}']
+
+    def writeconanscript(self, buildfolder):
+        conanparamsstr = ' '.join(self.current_buildparameters)
+        with open_script(Path(buildfolder) / "conanexport.sh") as f:
+            f.write('#!/bin/sh\n\n')
+            f.write(f'conan export-pkg . {self.libname}/{self.target_name} -f {conanparamsstr}\n')
+
+    def writeconanfile(self, buildfolder):
+        with (Path(buildfolder) / 'conanfile.py').open(mode='w', encoding='utf-8') as f:
+            f.write('from conans import ConanFile, tools\n')
```

Comment:
if it's any help at all you can do stuff like:
```py
f.write(f'''from conan import ConanFile, tools
class {self.libname}Conan(ConanFile):
    name = "{self.libname}"
    ...etc...
```

use a single multiline f-string. If you want to keep the indent there are things like https://docs.python.org/3/library/textwrap.html#textwrap.dedent too.